### PR TITLE
statistics: add `HotPeerStat.GetBytesRate()` & use float64 rates

### PR DESCRIPTION
**Pull request by Luffbee**
### What problem does this PR solve? 

* The `HotPeerStat` has a raw `BytesRate` and a denoised `RollingBytesRate`. There  are several places using the raw `BytesRate` should use the denoised `RollingBytesRate` actually.
* The type of rates like `BytesRate` and `KeysRate` should be `float64` instead of `uint64`.

### What is changed and how it works?

* Add `GetBytesRate()` for `HotPeerStat` to make it convenient to use `RollingBytesRate` and fix the places using wrong bytes rate.
* Change type of the rates to float64.

### Check List 

Tests 

 - Unit test

Code changes

 - Has HTTP API interfaces change

Related changes

 - Need to update the documentation

### CODE REVIEW

File: server/statistics/hot_peer_cache.go [hunk 1]
```diff
@@ -142,26 +142,26 @@ func (f *hotPeerCache) CollectMetrics(stats *StoresStats, typ string) {
 		store := storeTag(storeID)
 		threshold := f.calcHotThreshold(stats, storeID)
 		hotCacheStatusGauge.WithLabelValues("total_length", store, typ).Set(float64(peers.Len()))
-		hotCacheStatusGauge.WithLabelValues("hotThreshold", store, typ).Set(float64(threshold))
+		hotCacheStatusGauge.WithLabelValues("hotThreshold", store, typ).Set(threshold)
 	}
 }
 
-func (f *hotPeerCache) getBytesFlow(region *core.RegionInfo) uint64 {
+func (f *hotPeerCache) getBytesFlow(region *core.RegionInfo) float64 {
```
Comment (rleungx) [inline server/statistics/hot_peer_cache.go]: IMO, this function shouldn't be changed.

File: server/api/trend.go [hunk 2]
```diff
@@ -145,9 +145,9 @@ func (h *trendHandler) getStoreFlow(stats statistics.StoreHotRegionsStat, storeI
 		return
 	}
 	if stat, ok := stats[storeID]; ok {
-		storeFlow = stat.TotalBytesRate
+		storeFlow = uint64(stat.TotalBytesRate)
```
Comment (rleungx) [inline server/api/trend.go]: How about `storeFlow float64, regionFlows []float64` for return?

File: server/statistics/hot_peer_cache.go [hunk 3]
```diff
@@ -142,26 +142,26 @@ func (f *hotPeerCache) CollectMetrics(stats *StoresStats, typ string) {
...
+		return float64(region.GetBytesWritten())
 	case ReadFlow:
-		return region.GetBytesRead()
+		return float64(region.GetBytesRead())
 	}
 	return 0
 }
 
-func (f *hotPeerCache) getKeysFlow(region *core.RegionInfo) uint64 {
+func (f *hotPeerCache) getKeysFlow(region *core.RegionInfo) float64 {
```
Comment (rleungx) [inline server/statistics/hot_peer_cache.go]: ditto

File: server/api/trend.go (hunk 2 above)
Comment (Luffbee) [reply server/api/trend.go]: This function is only used by `/trend` API, and the API need uint64.

**Comment by Luffbee** [comment]
> Is this change compatible with TiDB? I see some declaim in `infoschema/tables`:
> 
> ```
> var tableTiDBHotRegionsCols = []columnInfo{
> 	{"TABLE_ID", mysql.TypeLonglong, 21, 0, nil, nil},
> 	{"INDEX_ID", mysql.TypeLonglong, 21, 0, nil, nil},
> 	{"DB_NAME", mysql.TypeVarchar, 64, 0, nil, nil},
> 	{"TABLE_NAME", mysql.TypeVarchar, 64, 0, nil, nil},
> 	{"INDEX_NAME", mysql.TypeVarchar, 64, 0, nil, nil},
> 	{"REGION_ID", mysql.TypeLonglong, 21, 0, nil, nil},
> 	{"TYPE", mysql.TypeVarchar, 64, 0, nil, nil},
> 	{"MAX_HOT_DEGREE", mysql.TypeLonglong, 21, 0, nil, nil},
> 	{"REGION_COUNT", mysql.TypeLonglong, 21, 0, nil, nil},
> 	{"FLOW_BYTES", mysql.TypeLonglong, 21, 0, nil, nil},
> }
> ```
> 
> The type is long long now.

No. I will change the PR description and API documents.

File: server/api/trend.go (hunk 2 above)
Comment (Luffbee) [reply server/api/trend.go]: Fixed.

File: server/statistics/hot_peer_cache.go (hunk 1 above)
Comment (Luffbee) [reply server/statistics/hot_peer_cache.go]: Fixed.

File: server/statistics/hot_peer_cache.go (hunk 3 above)
Comment (Luffbee) [reply server/statistics/hot_peer_cache.go]: Fixed

**Comment by Luffbee** [comment]
@nolouch most change didn't influenced API doc because of RAML do not support map (there are several FIXMEs in the doc).

**Comment by rleungx** [comment]
Why does this PR have a lot of changes in API docs?

**Comment by Luffbee** [comment]
> Why does this PR have a lot of changes in API docs?

The `api.html` is generated by `raml2html`. Seems there are some minor change in the theme. The real change of API docs is presented in `api.raml`, only 4 lines.

**Comment by rleungx** [comment]
> > Why does this PR have a lot of changes in API docs?
> 
> The `api.html` is generated by `raml2html`. Seems there are some minor change in the theme. The real change of API docs is presented in `api.raml`, only 4 lines.

I just ran the `scripts/build-api.sh` based on your PR, it modified the `docs/api.html` actually.

**Comment by Luffbee** [comment]
> > > Why does this PR have a lot of changes in API docs?
> > 
> > 
> > The `api.html` is generated by `raml2html`. Seems there are some minor change in the theme. The real change of API docs is presented in `api.raml`, only 4 lines.
> 
> I just ran the `scripts/build-api.sh` based on your PR, it modified the `docs/api.html` actually.

Fixed.

**Pull request merged**